src/Cli/dotnet/commands/dotnet-test/VSTestForwardingApp.cs

```diff
@@ -19,7 +21,10 @@ public VSTestForwardingApp(IEnumerable<string> argsToForward)
             if (!hasRootVariable)
             {
                 WithEnvironmentVariable(rootVariableName, rootValue);
+                VSTestTrace.WriteTrace($"Root variable set {rootVariableName}:{rootValue}");
             }
+
+            VSTestTrace.SafeWriteTrace(() => $"Forwarding to '{GetVSTestExePath()}' with args \"{argsToForward?.Aggregate((a, b) => $"{a} | {b}")}\"");
```

Comment:
Why `SafeWriteTrace` here and not at the other places?

Reply:
Ditto explained on the other comment.